src/logql-query.ts

```diff
@@ -0,0 +1,48 @@
+import { notEmptyString, trim } from './value-utils';
+
+export class LogQLQuery {
+  selectors: Array<string> = [];
+  pipeline: Array<string> = [];
+
+  constructor(query: string) {
+    const parts = query.match(/(.*){([^}]*)}(.*)/);
+    this.selectors =
+      parts && parts.length > 1
+        ? parts[2].split(',').filter(notEmptyString).map(trim)
+        : [];
+    this.pipeline =
+      parts && parts.length > 2
+        ? parts[3].split('|').filter(notEmptyString).map(trim)
+        : [];
+  }
+
+  appendSelector = (newSelectors: string | Array<string>) => {
+    if (Array.isArray(newSelectors)) {
+      this.selectors.push(...newSelectors);
+    } else {
+      this.selectors.push(newSelectors);
+    }
+
+    this.selectors = this.selectors.filter(notEmptyString).map(trim);
+
+    return this;
+  };
+
+  appendPipeline = (newPipeline: string | Array<string>) => {
+    if (Array.isArray(newPipeline)) {
+      this.pipeline = this.pipeline.concat(newPipeline);
```

Comment:
I notice we're using `concat` here, but `push(...newSelectors)` in `appendSelector`. Probably better to be consistent.

Reply:
yes missed that one while refactoring.